pyalgostrategypool/aaroon_crossover_v2.py

```diff
@@ -0,0 +1,152 @@
+import talib
+from constants import *
+from strategy.core.strategy_base import StrategyBase
+
+
+class AroonCrossover(StrategyBase):
+    name = 'Aroon Crossover v2'
+
+    def __init__(self, *args, **kwargs):
+        """
+        Accept and sanitize all your parameters here.
+        Set up the variables you will need here.
+        If you are running the strategy for multiple days, then this method will be called only once at the start of the strategy.
+        """
+
+        super().__init__(*args, **kwargs)
+
+        # Aroon parameter
+        self.time_period = self.strategy_parameters['TIME_PERIOD']
+
+        # Sanity
+        assert (0 < self.time_period == int(self.time_period)), f"Strategy parameter TIME_PERIOD should be a positive integer. Received: {self.time_period}"
+
+        # Variables
+        self.main_order_map = None
+
+    def initialize(self):
+        """
+        If you are running the strategy for multiple days, then this method will be called only once at the start of every day.
+        Use this to initialize and re-initialize your variables.
+        """
+        self.main_order_map = {}
+
+    def get_crossover_value(self, instrument):
+        """
+        This method calculates the crossover using the hist data of the instrument along with the required indicator and returns the entry/exit action.
+        """
+
+        # Get OHLC historical data for the instrument
+        hist_data = self.get_historical_data(instrument)
+
+        # Calculate the Aaroon values
+        aroon_down, aroon_up = talib.AROON(hist_data['high'], hist_data['low'], timeperiod=self.time_period)
+
+        # Get the crossover value
+        crossover_value = self.utils.crossover(aroon_up, aroon_down)
+
+        return crossover_value
+
+    def strategy_select_instruments_for_entry(self, candle, instruments_bucket):
+        """
+        This method is called once every candle time.
+        If you set the candle at 5 minutes, then this method will be called every 5 minutes (09:15, 09:20, 09:25 and so on).
+        In a candle, the exit method is called first, then the entry method is called.
+        So once a candle starts, strategy_select_instruments_for_exit gets called first
+        and then this method strategy_select_instruments_for_entry gets called.
+        """
+
+        # Add instrument in this bucket if you want to place an order for it,
+        # We decide whether to place an instrument in this bucket or not based on the decision-making process given below in the loop
+        selected_instruments = []
```

Comment:
selected_instruments = []
meta = []

change this to 
instruments, meta = [], []

do this in entry and exit for all strategies

Reply:
done